x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/rest/action/RestCatTransformAction.java

```diff
@@ -0,0 +1,231 @@
+                TableColumnAttributeBuilder.builder("exponential average checkpoint processing time (milliseconds)", false)
+                    .setAliases("cdtea", "checkpointTimeExpAvg")
+                    .build())
+            .addCell("indexed_documents_exp_avg",
+                TableColumnAttributeBuilder.builder("exponential average number of documents indexed", false)
+                    .setAliases("idea")
+                    .build())
+            .addCell("processed_documents_exp_avg",
+                TableColumnAttributeBuilder.builder("exponential average number of documents processed", false)
+                    .setAliases("pdea")
+                    .build())
+            .endHeaders();
+    }
+
+    private Table buildTable(GetTransformAction.Response response, GetTransformStatsAction.Response statsResponse) {
+        Table table = getTableWithHeader();
+        Map<String, TransformStats> statsById = statsResponse.getTransformsStats().stream()
+                                                                .collect(Collectors.toMap(TransformStats::getId, Function.identity()));
+        response.getTransformConfigurations().forEach(config -> {
+            TransformStats stats = statsById.get(config.getId());
+            TransformCheckpointingInfo checkpointingInfo = null;
+            TransformIndexerStats transformIndexerStats = null;
+
+            if(stats != null) {
+                checkpointingInfo = stats.getCheckpointingInfo();
+                transformIndexerStats = stats.getIndexerStats();
+            }
+
+            table
+                .startRow()
+                .addCell(config.getId())
+                .addCell(config.getCreateTime())
+                .addCell(config.getVersion())
+                .addCell(String.join(",", config.getSource().getIndex()))
+                .addCell(config.getDestination().getIndex())
+                .addCell(config.getDescription())
+                .addCell(config.getSyncConfig() == null ? "batch" : "continuous")
+                .addCell(config.getFrequency() == null ? "one-time" : config.getFrequency())
+                .addCell(stats == null ? null : stats.getState())
+                .addCell(stats == null ? null : stats.getReason())
+                .addCell(checkpointingInfo == null ? null : checkpointingInfo.getChangesLastDetectedAt())
+                .addCell(transformIndexerStats == null ? null : transformIndexerStats.getSearchTotal())
+                .addCell(transformIndexerStats == null ? null : transformIndexerStats.getSearchFailures())
+                .addCell(transformIndexerStats == null ? null : transformIndexerStats.getSearchTime())
+
+                .addCell(transformIndexerStats == null ? null : transformIndexerStats.getIndexTotal())
+                .addCell(transformIndexerStats == null ? null : transformIndexerStats.getIndexFailures())
+                .addCell(transformIndexerStats == null ? null : transformIndexerStats.getIndexTime())
```

Comment:
```suggestion
                .addCell(transformIndexerStats == null ? null : TimeValue.timeValueMillis(transformIndexerStats.getIndexTime()))
```

Reply:
Done.

I see that for `transformIndexerStats.getExpAvgCheckpointDurationMs()` below, it is returning a `double` object, but `TimeValue#timeValueMillis` only takes in `long`. Should I update it as well at the cost of losing precision by  converting `double` to `long`, or it's fine to leave it as is?

Reply:
I would leave the double as is. Just like you did :). 

It might be interesting to have factional `TimeValue` objects in the future. So we could get human friendly output like `10.8ms` :D.